swagger.yaml

```diff
@@ -90,87 +90,94 @@ definitions:
     description: The top-level node of a hierarchy
     type: object
     properties:
+      children:
+        description: The children of the top-level node of this hierarchy
+        type: array
+        items:
+          $ref: '#/definitions/NodeRef'
+      has_data:
+        description: True if the instance has an observation for this code
+        type: boolean
       label:
         $ref: '#/definitions/Label'
       links:
         $ref: '#/definitions/Links'
       no_of_children:
-        type: integer
         description: The number of child nodes that this node has
-      has_data:
-        type: boolean
-        description: True if the instance has an observation for this code
-      children:
-        type: array
-        description: The children of the top-level node of this hierarchy
-        items:
-          $ref: '#/definitions/NodeRef'
+        type: integer
   Links:
     description: A list of links related to this node
     readOnly: true
     type: object
     properties:
-      self:
-        $ref: '#/definitions/Link'
       code:
         $ref: '#/definitions/Link'
-    example:
       self:
-        id: abc123
-        href: 'http://hierarchy-api/hierarchies/instance_id1/dimension1'
+        $ref: '#/definitions/SelfLink'
+    example:
       code:
-        id: xyz987
         href: 'http://codelist/code-lists/clist1/codes/xyz987'
+        id: xyz987
+      self:
+        href: 'http://hierarchy-api/hierarchies/instance_id1/dimension1'
   CodeResponse:
     description: ''
     type: object
     properties:
-      links:
-        $ref: '#/definitions/Links'
-      label:
-        type: string
-        description: The label for this node
-      no_of_children:
-        type: integer
-        description: The number of child nodes that this node has
-      has_data:
-        type: boolean
-        description: True if the instance has an observation for this code
-      children:
-        type: array
-        description: The child nodes of this node in the hierarchy
-        items:
-          $ref: '#/definitions/NodeRef'
       breadcrumbs:
-        type: array
         description: >-
           The ancestors (nodes) of this node in the hierarchy, starting with the
           parent
+        type: array
+        items:
+          $ref: '#/definitions/NodeRef'
+      children:
+        description: The child nodes of this node in the hierarchy
         items:
           $ref: '#/definitions/NodeRef'
+        type: array
+      has_data:
+        description: True if the instance has an observation for this code
+        type: boolean
+      label:
+        description: The label for this node
+        type: string
+      links:
+        $ref: '#/definitions/Links'
+      no_of_children:
+        description: The number of child nodes that this node has
+        type: integer
   NodeRef:
-    type: object
+    description: ""
```

Comment:
why so blank?

Reply:
I didn't know what the description should be yesterday. Having a second look today I realise that the NodeRef is used by both root node and child nodes and therefore would need different descriptions.

The different descriptions are already in spec against the children field (type array) on lines 94 and 128; hence this property can be removed.